php/libraries/ValidatableIdentifier.php

```diff
@@ -0,0 +1,65 @@
+<?php declare(strict_types=1);
+/**
+ * Implementation of ValidatableIdentifier needs to provide a validation
+ * function to ensure that no Identifier of that type exists if the value is
+ * not validated.
+ *
+ * PHP Version 7
+ *
+ * @category StudyEntities
+ * @package  LORIS
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+/**
+ * ValidatableIdentifier with the default constructor that call validate
+ *
+ * @category StudyEntities
+ * @package  LORIS
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+abstract class ValidatableIdentifier implements Identifier
+{
+    /**
+     * The Identifier's value
+     *
+     * @var int|string The identifier value to be validated
```

Comment:
It looks like this can only be a string since it's set by the constructor, which takes a string (which means that __toString can be moved up a level from SessionID to ValidatableIdentifier)

Reply:
You're right, I left this in by mistake from an earlier idea.